pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -2720,25 +2746,55 @@ private void updatePolicies(NamespaceName ns, Function<Policies, Policies> updat
        }
    }
 
-    protected void internalSetNamespaceResourceGroup(String rgName) {
-        validateNamespacePolicyOperation(namespaceName, PolicyName.RESOURCEGROUP, PolicyOperation.WRITE);
-        validatePoliciesReadOnlyAccess();
-
-        if (rgName != null) {
-            // check resourcegroup exists.
-            try {
-                if (!resourceGroupResources().resourceGroupExists(rgName)) {
-                    throw new RestException(Status.PRECONDITION_FAILED, "ResourceGroup does not exist");
-                }
-            } catch (Exception e) {
-                log.error("[{}] Invalid ResourceGroup {}: {}", clientAppId(), rgName, e);
-                throw new RestException(e);
-            }
-        }
 
-        internalSetPolicies("resource_group_name", rgName);
+    protected void internalGetNamespaceResourceGroup(AsyncResponse asyncResponse, String tenant, String namespace) {
+        validateNamespacePolicyOperationAsync(NamespaceName.get(tenant, namespace),
+                PolicyName.RESOURCEGROUP, PolicyOperation.READ)
+                .thenCompose(__ -> getNamespacePoliciesAsync(namespaceName)
+                        .thenAccept(policies -> asyncResponse.resume(policies.resource_group_name)))
```

Comment:
```suggestion
                        .whenComplete((policies,ex) -> {
                          if (ex == null) {
                            log.info("[{}] Successfully to get namespace resource group {}/{}", clientAppId(), tenant, namespace);
                            asyncResponse.resume(policies.resource_group_name)
                            return;
                          }
                          Throwable realCause = FutureUtil.unwrapCompletionException(ex);
                          log.error("[{}] Fail to get namespace resource group {}/{}: {}", clientAppId(), tenant, namespace, realCause);
                          resumeAsyncResponseExceptionally(asyncResponse, realCause);
                          return;
                        ))
```

Reply:
I think we don't need to add ``whenComplete `` to increase code complexity. So I add successful log info in ``thenAccept``.